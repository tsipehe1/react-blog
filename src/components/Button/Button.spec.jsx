import { render, screen } from '@testing-library/react';
import { Button } from '.';
import userEvent from '@testing-library/user-event';

describe('<Button />', () => {
  it('sould render the button with the text "Load more"', () => {
    const fn = jest.fn(); //mock a function

    render(<Button text="Load more" onClick={fn} />);
    expect.assertions(1); // para ter a certeza em test async que a asserÃ§ao "expect" occorreu

    const button = screen.getByRole('button', { name: /load more/i });

    expect(button).toBeInTheDocument();
  });

  it('sould call function on button click', () => {
    const fn = jest.fn(); //mock a function

    render(<Button text="Load more" onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    userEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('sould be disabled when disabled is true', () => {
    const fn = jest.fn(); //mock a function

    render(<Button text="Load more" disabled={true} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    userEvent.click(button);

    expect(button).toBeDisabled();
  });

  it('sould be enabled when disabled is false', () => {
    const fn = jest.fn(); //mock a function

    render(<Button text="Load more" disabled={false} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    userEvent.click(button);

    expect(button).toBeEnabled();
  });

  it('sould should match snapshot', () => {
    const fn = jest.fn(); //mock a function

    const { container } = render(<Button text="Load more" disabled={false} onClick={fn} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
