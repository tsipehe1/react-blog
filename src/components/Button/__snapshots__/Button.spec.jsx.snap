// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> sould should match snapshot 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
